pkg/state/state_skel.go

```diff
@@ -106,6 +173,13 @@ func (s *stateSkel) createOrUpdateObjs(
 			return errors.Wrap(err, "failed to set controller reference for object")
 		}
 
+		labels := desiredObj.GetLabels()
```

Comment:
can you add comment here ?
// add state specific labels 

or move it to separate function : `addStateSpecificLabels(obj)`  ?

Reply:
done